### ISSUE TITLE
Implement all task pipelines

### ISSUE THREAD

**Issue by atreyasha** [infsys-lab/privacy-glue#5]
# Training scripts

Branch off into 3 unique training *scripts* which are called by `privacy_glue.py`. They should execute training and ultimately dump all metrics and logs. Dumped data should be as clean as possible. They do not need to execute parsing and other initialization workflows already executed in `privacy_glue.py`.

- [x] `sequence_classification.py` for OPP-115, Policy-Detection, PolicyIE-A and PrivacyQA (@Christof93)
- [x] `wandb` to store results and models (@Christof93)  
- [x] `sequence_tagging.py` for PI-Extract and PolicyIE-B using one encoder and two heads (@Christof93)
- [x] `reading_comprehension.py` for PolicyQA (@atreyasha)
- [x] Unit testing for all scripts and low performance insights for OPP-115 and Sequence Tagging (@atreyasha)

These should be adapted from https://github.com/huggingface/transformers/tree/main/examples/pytorch with better unit and integration testing. Note that we define shared arguments in `parser.py`, so please modify that accordingly.

For multi-label classification, use either `problem_type` parameter for supported models or write your own `compute_loss` function as described below:

https://discuss.huggingface.co/t/most-efficient-multi-label-classifier/9296/2
https://discuss.huggingface.co/t/multi-label-token-classification/16509/6

Once good, we submit PRs to merge these scripts to the `main` branch with mutual reviews.

# Review pointers

## Data loading

- [x] Add feature information attribute for all Sequence Classification and Sequence Tagging tasks so these don't need to computed in the pipeline
- [x] Remove feature information attributes in the pipeline if they were already there

## Optimizations

- [x] Always read latest checkpoint and share across as `self.last_checkpoint` variable -> ignore `train_args.resume_from_checkpoint` variable
- [x] Implement training optimizations such as early stopping for all tasks and weighted random samplers where relevant
- [x] Resume training should be done with `save_strategy="epoch"` and `metric_for_best_model` and `greater_is_better` should be explicitly defined within pipeline per class
- [x] Define batch size and other custom variables per pipeline if necessary

## Evaluation

- [x] Besides metrics, raw predictions should be dumped for post-hoc analysis

## Organization

- [x] Delete old checkpoints when `do_clean` is passed
- [x] Standardize usage of `self.trainer.is_world_process_zero` and `self.train_args.main_process_first` across pipelines
- [x] Merge utilities in the `utils` directory for easy re-use across source code
- [x] ~Rename abstract methods to more meaningful names eg. `_retrieve_data` to something else -> also consider moving `_retrieve_data` into common pipeline method since this is always called~
- [x] Consistently use f-strings
